sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/UnnecessaryImportsCheck.kt

```diff
@@ -141,23 +150,68 @@ class UnnecessaryImportsCheck : AbstractCheck() {
                 }
             }
         }
+
+    private fun getDelegatesImportsFilter(propertyDelegates: Collection<KtPropertyDelegate>, bindingContext: BindingContext) =
+        if (bindingContext == BindingContext.EMPTY) {
+            { imp: KtImportDirective ->
+                imp.importedName?.asString().let { it != "getValue" && it != "setValue" && it != "provideDelegate"}
+            }
+        } else {
+            propertyDelegates.flatMap { propDelegate ->
+                bindingContext.get(BindingContext.DELEGATE_EXPRESSION_TO_PROVIDE_DELEGATE_CALL, propDelegate.expression)
+                    .getResolvedCall(bindingContext)
+                    ?.resultingDescriptor
+                    ?.fqNameOrNull()?.let { listOf(it) }
+                    ?: (propDelegate.parent as? KtProperty)?.let { ktProperty ->
+                        (bindingContext.get(BindingContext.DECLARATION_TO_DESCRIPTOR, ktProperty) as? PropertyDescriptor)?.let { propDescriptor ->
+                            propDescriptor
+                                .accessors
+                                .mapNotNull {
+                                    bindingContext.get(BindingContext.DELEGATED_PROPERTY_RESOLVED_CALL, it)
+                                        ?.resultingDescriptor
+                                        ?.fqNameOrNull()
+                                }
+                        }
+                    } ?: emptyList()
+            }.let { resolvedArrayAccesses ->
+                { imp: KtImportDirective ->
+                    imp.importedFqName !in resolvedArrayAccesses
+                }
+            }
+        }
+
+    private fun getInvokeCallsImportsFilter(calls: Collection<KtCallExpression>, bindingContext: BindingContext) =
+        if (bindingContext == BindingContext.EMPTY) {
+            { imp: KtImportDirective -> imp.importedName?.asString() != "invoke" }
+        } else {
+            { imp: KtImportDirective ->
+                if (imp.importedName?.asString() != "invoke") true
+                else {
+                    imp.importedFqName !in calls.mapNotNull { it.getResolvedCall(bindingContext)?.resultingDescriptor?.fqNameOrNull() }
+                }
+            }
+        }
 }
 
-private class DataCollector(val file: KtFile) : KtTreeVisitorVoid() {
+private class DataCollector(val file: KtFile, val bindingContext: BindingContext) : KtTreeVisitorVoid() {
```

Comment:
I think you don't need the `bindingContext` here (and correspondingly in the method that instantiates the `DataCollector`)